orttraining/orttraining/training_ops/cpu/torch/torch_custom_function_kernel_base.cc

```diff
@@ -258,11 +258,11 @@ void PythonOpGradBase::RunBackward(OpKernelContext* context,
 
 void PythonOpGradBase::SetOutputs(OpKernelContext* context, std::vector<OrtValue>& returned_ortvalues) const {
   auto* ctx_internal = reinterpret_cast<onnxruntime::OpKernelContextInternal*>(context);
-  auto outputs_count = static_cast<size_t>(ctx_internal->OutputCount());
-  // It's possible that Pytorch returns None as gradient and ORT Python side may skip them.
-  // In that case, returned_args may contain less arguments.
-  outputs_count = outputs_count > returned_ortvalues.size() ? returned_ortvalues.size() : outputs_count;
+  const auto outputs_count = static_cast<size_t>(ctx_internal->OutputCount());
```

Comment:
seems our cases all assume None happens at the end. E.g. value1, value2, None, None

maybe we should add cases for forward/backward, having outputs like value1, None, value2, None, None, ...,value3.
what do you think?

Reply:
Yep. It's already a part of my local changes you saw today.

Reply:
I will add a test seperately.